### PR TITLE
[main][feat][alpaca-7788]: script for deploy money market at arbiturm

### PR DESCRIPTION
**Pull request by sundayalpaca**
## Description

- script for deploy money market at arbiturm

### CODE REVIEW

**Review by 69alpacas — APPROVED** [review]

File: .arbitrum.mainnet.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      "routerV3": "",
+      "factoryV3": ""
+    },
+    "thenaDex": {
+      "routerV2": "",
+      "routerV3": ""
+    }
+  },
+  "moneyMarket": {
+    "moneyMarketDiamond": "0x588c58d88319B2EDF7426006668cDfF60940F3C7",
```
Comment (jr-alpaca) [inline .arbitrum.mainnet.json:45]: please lease undeployed contracts as empty string

File: .openzeppelin/unknown-42161.json [hunk 2]
```diff
@@ -0,0 +1,419 @@
+{
```
Comment (jr-alpaca) [inline .openzeppelin/unknown-42161.json:1]: don't push this file. It is for deployer only.

File: hardhat.config.ts [hunk 3]
```diff
@@ -15,6 +15,11 @@ const config: HardhatUserConfig = {
       url: process.env.BSC_RPC_URL,
       accounts: process.env.DEPLOYER_PRIVATE_KEY !== undefined ? [process.env.DEPLOYER_PRIVATE_KEY] : [],
     },
+    arbitrum_mainnet: {
+      chainId: 42161,
+      url: process.env.ARB_RPC_URL,
```
Comment (jr-alpaca) [inline hardhat.config.ts:20]: please add ARB_RPC_URL in env.example
```suggestion
      url: process.env.ARB_RPC_URL,
```

File: deploy/money-market/config/set-repurchase-reward-model.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  Check all variables below before execute the deployment script
+  */
+
+  const fixFeeModel500Bps = config.sharedConfig.fixFeeModel500Bps;
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setRepurchaseRewardModel : ${fixFeeModel500Bps}`);
+
+  await iMoneyMarketFactory.setRepurchaseRewardModel(fixFeeModel500Bps);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-repurchase-reward-model.ts:28]: please log transaction hash.

File: deploy/money-market/config/set-risk-manager.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  */
+
+  const isOk = true;
+  const riskManagers = ["0xC44f82b07Ab3E691F826951a6E335E1bC1bB0B51"];
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setRiskManagersOk : ${riskManagers} , ${isOk}`);
+
+  await iMoneyMarketFactory.setRiskManagersOk(riskManagers, isOk);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-risk-manager.ts:29]: please log transaction hash.

File: deploy/money-market/config/set-oracle.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  Check all variables below before execute the deployment script
+  */
+
+  const newOracle = config.oracle.alpacaV2Oracle02;
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setOracle : ${newOracle}`);
+
+  await iMoneyMarketFactory.setOracle(newOracle);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-oracle.ts:28]: please log transaction hash.

File: deploy/money-market/config/set-min-debt-size.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  Check all variables below before execute the deployment script
+  */
+
+  const minDebtSize = ethers.utils.parseEther("100");
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setMinDebtSize : ${minDebtSize.toString()}`);
+
+  await iMoneyMarketFactory.setMinDebtSize(minDebtSize);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-min-debt-size.ts:29]: please log transaction hash.

File: deploy/money-market/config/set-max-num-of-token.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  const numOfCollat = 10;
+  const numOfDebt = 10;
+  const numOfNonCollatDebt = 0;
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setMaxNumOfToken : ${numOfCollat} , ${numOfDebt} , ${numOfNonCollatDebt}`);
+
+  await iMoneyMarketFactory.setMaxNumOfToken(numOfCollat, numOfDebt, numOfNonCollatDebt);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-max-num-of-token.ts:30]: please log transaction hash.

File: deploy/money-market/config/set-liquidation-treasury.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  Check all variables below before execute the deployment script
+  */
+
+  const treasury = "0xFeCfcd99B496e044166086dd2F29E2FC2bb6Dd64";
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setLiquidationTreasury : ${treasury}`);
+
+  await iMoneyMarketFactory.setLiquidationTreasury(treasury);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-liquidation-treasury.ts:28]: please log transaction hash.

File: deploy/money-market/config/set-liquidation-param.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  */
+
+  const newMaxLiquidateBps = 5000;
+  const newLiquidationThreshold = 11000;
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setLiquidationParams : ${newMaxLiquidateBps} , ${newLiquidationThreshold}`);
+
+  await iMoneyMarketFactory.setLiquidationParams(newMaxLiquidateBps, newLiquidationThreshold);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-liquidation-param.ts:29]: please log transaction hash.

File: deploy/money-market/config/set-ib-token-implementation.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  Check all variables below before execute the deployment script
+  */
+
+  const ibTokenImplementation = config.moneyMarket.interestBearingTokenImplementation;
+
+  const iMoneyMarketFactory = await IMoneyMarket__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  console.log(`> 🟢 setIbTokenImplementation : ${ibTokenImplementation}`);
+
+  await iMoneyMarketFactory.setIbTokenImplementation(ibTokenImplementation);
```
Comment (jr-alpaca) [inline deploy/money-market/config/set-ib-token-implementation.ts:28]: please log transaction hash.

File: deploy/money-market-account-manager/deploy/money-market-account-manager.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+  ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+  ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+  ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+  ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+  Check all variables below before execute the deployment script
+  */
+
+  const moneyMarket = config.moneyMarket.moneyMarketDiamond;
+  const weth = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1";
```
Comment (jr-alpaca) [inline deploy/money-market-account-manager/deploy/money-market-account-manager.ts:23]: get address from config?
```suggestion
  const weth = config.tokens.weth;
```

File: deploy/core/facets/diamond-cut/execute-diamond-cut-view.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    }),
+  },
+];
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const deployer = await getDeployer();
+
+  const poolDiamond = MMDiamondCutFacet__factory.connect(config.moneyMarket.moneyMarketDiamond, deployer);
+
+  await (await poolDiamond.diamondCut(facetCuts, ethers.constants.AddressZero, "0x")).wait();
```
Comment (jr-alpaca) [inline deploy/core/facets/diamond-cut/execute-diamond-cut-view.ts:34]: please log transaction hash.

File: .arbitrum.mainnet.json [hunk 14]
```diff
@@ -0,0 +1,84 @@
+{
+  "proxyAdmin": "",
+  "nativeRelayer": "",
+  "timelock": "",
+  "opMultiSig": "",
+  "usdPlaceholder": "",
+  "tokens": {
+    "weth": ""
```
Comment (jr-alpaca) [inline .arbitrum.mainnet.json:8]: no need to remove weth token
```suggestion
    "weth": ""
```

**Pull request closed without merging**